my_game.py

```diff
@@ -128,14 +129,36 @@ def thrust(self):
             self.change_x *= player_x_and_y_speed_ratio
             self.change_y *= player_x_and_y_speed_ratio
 
-    def update(self):
+    def reset(self):
+        self.reset_timer = PLAYER_INVINCIBILTY_SECONDS
+        # The Player is Invisible
+        self.alpha = 0
+
+    @property
+    def invicible(self):
+        return self.reset_timer < 1
+
+    def update(self, delta_time):
```

Comment:
Please rename is to _on_update()_. That is the standard name for the update function with argument _delta_time_.